frontend/src/bundles/common/components/button/button.tsx

```diff
@@ -1,19 +1,57 @@
-import { Button } from '@mui/material';
+import {
+    Button,
+    type ButtonPropsColorOverrides,
+    type SxProps,
+    type Theme,
+} from '@mui/material';
 
 type Properties = {
     label: string;
     variant?: 'text' | 'outlined' | 'contained';
     type?: 'submit' | 'button';
+    disabled?: boolean;
+    className?: string;
+    sx?: SxProps<Theme>;
+    endIcon?: React.ReactNode;
+    startIcon?: React.ReactNode;
+    color?:
+        | 'inherit'
+        | 'primary'
+        | 'secondary'
+        | 'success'
+        | 'error'
+        | 'info'
+        | ('warning' & ButtonPropsColorOverrides)
+        | undefined;
+    onClick?: React.MouseEventHandler<HTMLButtonElement>;
 };
 
 const CustomButton: React.FC<Properties> = ({
```

Comment:
I would leave the component name the same as the filename.

```suggestion
const Button: React.FC<Properties> = ({
```